### PR TITLE
Compat fixes

### LINKED ISSUE #123: Example doesn't work
```
import urllib
from tqdm import tqdm

def my_hook(t):
  """
  Wraps tqdm instance. Don't forget to close() or __exit__()
  the tqdm instance once you're done with it (easiest using `with` syntax).

  Example
  -------

  >>> with tqdm(...) as t:
  ...     reporthook = my_hook(t)
  ...     urllib.urlretrieve(..., reporthook=reporthook)

  """
  last_b = [0]

  def inner(b=1, bsize=1, tsize=None):
    """
    b  : int, optional
        Number of blocks just transferred [default: 1].
    bsize  : int, optional
        Size of each block (in tqdm units) [default: 1].
    tsize  : int, optional
        Total size (in tqdm units). If [default: None] remains unchanged.
    """
    if tsize is not None:
        t.total = tsize
    t.update((b - last_b[0]) * bsize)
    last_b[0] = b
  return inner

eg_link = 'http://www.doc.ic.ac.uk/~cod11/matryoshka.zip'
with tqdm(unit='B', unit_scale=True, leave=True, miniters=1,
          desc=eg_link.split('/')[-1]) as t:  # all optional kwargs
    urllib.urlretrieve(eg_link, filename='/dev/null',
                       reporthook=my_hook(t), data=None)
```

Traceback:
```
ValueError                                Traceback (most recent call last)
<ipython-input-29-e2aca7a860c8> in <module>()
     36           desc=eg_link.split('/')[-1]) as t:  # all optional kwargs
     37     urllib.urlretrieve(eg_link, filename='/dev/null',
---> 38                        reporthook=my_hook(t), data=None)

/usr/lib/python2.7/urllib.pyc in urlretrieve(url, filename, reporthook, data, context)
     96     else:
     97         opener = _urlopener
---> 98     return opener.retrieve(url, filename, reporthook, data)
     99 def urlcleanup():
    100     if _urlopener:

/usr/lib/python2.7/urllib.pyc in retrieve(self, url, filename, reporthook, data)
    269                     size = int(headers["Content-Length"])
    270                 if reporthook:
--> 271                     reporthook(blocknum, bs, size)
    272                 while 1:
    273                     block = fp.read(bs)

<ipython-input-29-e2aca7a860c8> in inner(b, bsize, tsize)
     28     if tsize is not None:
     29         t.total = tsize
---> 30     t.update((b - last_b[0]) * bsize)
     31     last_b[0] = b
     32   return inner

/usr/local/lib/python2.7/dist-packages/tqdm/_tqdm.pyc in update(self, n)
    532 
    533         if n < 1:
--> 534             raise ValueError("n ({0}) cannot be less than 1".format(n))
    535         self.n += n
    536 

ValueError: n (0) cannot be less than 1
```

### PR DESCRIPTION
**Pull request by casperdcl**
- Fixes #123
- may help fix fake streams and stdout wrappers (see #122)
- how-to import without enforcing dependency example

### CODE REVIEW

File: tqdm/_utils.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 try:  # pragma: no cover
     import colorama
     colorama.init()
-except ImportError:  # pragma: no cover
+except:  # pragma: no cover
```
Comment (lrq3000) [inline tqdm/_utils.py]: Why remove the ImportError and catch all kinds of exceptions?

**Comment by lrq3000** [comment]
Thank's @casperdcl. And good idea about #122, I hope this will fix the issue :S

File: tqdm/_utils.py (hunk 1 above)
Comment (lrq3000) [inline tqdm/_utils.py]: After discussion in #126, I guess this is the fix you mentioned @casperdcl. Ok I agree with the change, but I think that we should print a warning if the exception is different than ImportError to be easier for us to debug in the future.

Comment (casperdcl) [inline tqdm/_utils.py]: reverted this because of colorama update

**Pull request merged**